tests/cleanup/remove_old_objects_openstack.sh

```diff
@@ -0,0 +1,121 @@
+        fi
+
+        if [ -z "$OS_USERNAME" ]; then
+            rlFail "OS_USERNAME is empty!"
+        else
+            rlLogInfo "OS_USERNAME is configured"
+        fi
+
+        if [ -z "$OS_PASSWORD" ]; then
+            rlFail "OS_PASSWORD is empty!"
+        else
+            rlLogInfo "OS_PASSWORD is configured"
+        fi
+
+        # VMs older than HOURS_LIMIT will be deleted
+        HOURS_LIMIT="${HOURS_LIMIT:-24}"
+        export TIMESTAMP=`date -u -d "$HOURS_LIMIT hours ago" '+%FT%T'`
+
+        rlLogInfo "HOURS_LIMIT=$HOURS_LIMIT"
+
+        for package in ansible python3-openstacksdk python3-openstackclient; do
+            if ! rlCheckRpm "$package"; then
+                rlRun -t -c "dnf -y install $package"
+                rlAssertRpm "$package"
+            fi
+        done
+        PLAYBOOK_DELETE_VMS=`mktemp`
+        PLAYBOOK_DELETE_IMAGES=`mktemp`
+    rlPhaseEnd
+
+    rlPhaseStartTest "Delete old VMs"
+        cat > $PLAYBOOK_DELETE_VMS << __EOF__
+- name: Delete old VMs
+  hosts: localhost
+  gather_facts: False
+  tasks:
+  - name: Get VMs
+    os_server_facts:
+
+  - name: List all VMs
+    debug:
+      var: openstack_servers
+
+  - name: Delete old VMs
+    os_server:
+      name: "{{item.id}}"
+      state: absent
+    loop: "{{openstack_servers}}"
+    when: item.created < lookup('env','TIMESTAMP')
+    loop_control:
+      label: "{{item.name}} (id: {{item.id}} created: {{item.created}})"
+__EOF__
+
+        rlLogInfo "Removing VMs created before $TIMESTAMP"
+        rlRun -t -c "ansible-playbook $PLAYBOOK_DELETE_VMS"
+    rlPhaseEnd
+
+    rlPhaseStartTest "Delete old images"
+        cat > $PLAYBOOK_DELETE_IMAGES << __EOF__
+- name: Delete old images
+  hosts: localhost
+  gather_facts: False
+  tasks:
+  - name: Get images
+    os_image_facts:
+
+# listing details about all images can produce tens of thousands of lines, so don't do it
+#  - name: List all images
+#    debug:
+#      var: openstack_image
+
+  - name: Delete old images
+    os_image:
+      name: "{{item.name}}"
+      id: "{{item.id}}"
+      state: absent
+    loop: "{{openstack_image}}"
+    when: (item.created_at < lookup('env','TIMESTAMP')) and (item.name | regex_search('Composer-[a-f0-9-]{36}-Automated-Import'))
+    loop_control:
+      label: "{{item.name}} (id: {{item.id}} created: {{item.created_at}})"
+__EOF__
+
+        rlLogInfo "Removing images created before $TIMESTAMP"
+        rlRun -t -c "ansible-playbook $PLAYBOOK_DELETE_IMAGES"
+    rlPhaseEnd
+
+    rlPhaseStartTest "Delete old volumes"
+        volume_list=`openstack-3 volume list --format value --column ID`
+        for volume in $volume_list; do
+            creation_date=`openstack-3 volume show $volume --column created_at --format value`
+            if [ $? -ne 0 ]; then
+                rlLogWarning "Failed to get the creation date of volume $volume"
+                continue
+            fi
+            if [ "$creation_date" \< "$TIMESTAMP" ]; then
```

Comment:
What date/time format is this? Are we certain that comparing the dates as strings will compare them in chronological order ?

Also why `\<` ? Isn't `<` enough ?

Reply:
OK, this is important to know. 

Can you add this comment in the source code. Maybe same for the `openstack_image` and `openstack_servers` variables so that when somebody else reads this script later they will know where these things are coming from and why they are written in this way.

Other than the comments this is ready to be merged IMO.

Reply:
Comments have been added in the latest version of the script.